src/beacon/app/routes/analytics.py

```
@@ -0,0 +1,481 @@
+                DeviceReading.device_key.in_(device_keys),
+                DeviceReading.timestamp >= start_date,
+                DeviceReading.timestamp <= end_date
+            )
+        ).first()
+        
+        valid_readings = db.exec(
+            select(func.count(DeviceReading.reading_key)).where(
+                DeviceReading.device_key.in_(device_keys),
+                DeviceReading.timestamp >= start_date,
+                DeviceReading.timestamp <= end_date,
+                (DeviceReading.s1_pm2_5 != None) | (DeviceReading.s2_pm2_5 != None)
```

Ruff (0.12.2)

Comparison to `None` should be `cond is not None`

Replace with `cond is not None`

(E711)